src/main/scala/edu/ie3/powerFactory2psdm/util/GridPreparator.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * © 2021. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+
+package edu.ie3.powerFactory2psdm.util
+
+import com.typesafe.scalalogging.LazyLogging
+import edu.ie3.powerFactory2psdm.model.powerfactory.PowerFactoryGrid
+import edu.ie3.powerFactory2psdm.model.powerfactory.PowerFactoryGrid.Switches
+
+object GridPreparator extends LazyLogging {
+
+  def isFullyConnectedSwitch(switch: Switches): Boolean =
+    switch.bus1Id.isDefined & switch.bus2Id.isDefined
+
+  /**
+    * Removes [[Switches]] from a [[PowerFactoryGrid]] that are only connected to a single node.
+    * @param maybeSwitches
+    * @return
+    */
+  def removeSinglyConnectedSwitches(
+      maybeSwitches: Option[List[PowerFactoryGrid.Switches]]
+  ): Option[List[PowerFactoryGrid.Switches]] = maybeSwitches match {
+    case Some(switches) =>
+      val (fullyConnected, singlyConnected) =
+        switches.partition(isFullyConnectedSwitch)
+      singlyConnected.foreach(
+        switch =>
+          logger.debug(
+            s"Removed switch with id: ${switch.id.getOrElse("NO_ID")}, since it only has a single connection."
+          )
+      )
+      Some(fullyConnected)
+  }
+
+  /**
+    * Perform preparation of the [[PowerFactoryGrid]] before the actual conversion can happen.
+    *
+    * @param pfGrid the [[PowerFactoryGrid]] to prepare
+    * @return the prepared [[PowerFactoryGrid]]
+    */
+  def prepare(pfGrid: PowerFactoryGrid): PowerFactoryGrid = {
```

Comment:
Regarding the clean code paradigm, you typically structure classes and objects like a waterfall. So the public methods etc. are upfront and then the rest follows as they are called. For your object, the clean way would be:

1. `prepare`
2. `removeSinglyConnectedSwitches`
3. `isFullyConnectedSwitch` 

It is to aid the user / other developers, because they can "read" the class from top to buttom.

Reply:
Makes sense. Thanks for the hint!!